### DISCUSSION TITLE
✅ August 2024 (v0.5)

### DISCUSSION THREAD

**Discussion by flanakin** [microsoft/finops-toolkit#818] [Releases]
### 🆕 New in August 2024
- 📗 [Implementing FinOps guide](https://microsoft.github.io/finops-toolkit/#-learning-resources) has a new doc to compare FOCUS and actual/amortized datasets.
- 📊 [Power BI reports](https://aka.ms/ftk/pbi) now support raw Cost Management exports and removed the dependency on the Cost Management connector.
- 📒 [Cost optimization workbook](https://microsoft.github.io/finops-toolkit/optimization-workbook) includes new and updated queries.
- 🔍 [Azure Optimization Engine](https://microsoft.github.io/finops-toolkit/optimization-engine) added a new Register-MultitenantAutomationSchedules script, ZRS disks for underutilized disk recommendations, and an option to scope exports to MCA billing profiles.

### ℹ️ Updated
- 🏦 [FinOps hubs](https://aka.ms/finops/hubs) had various small fixes and improvements.
- 📊 [Power BI reports](https://aka.ms/ftk/pbi) had various small fixes and improvements across all reports.
- 📒 [Governance workbook](https://microsoft.github.io/finops-toolkit/governance-workbook) improved the subscription filtering experience.
- 🔍 [Azure Optimization Engine](https://microsoft.github.io/finops-toolkit/optimization-engine) improved SQL database security and various other small fixes and improvements.
- 🖥️ [FinOps toolkit PowerShell commands](https://aka.ms/ftk/ps) had various small fixes and improvements to export and FinOps hub management commands.
- 🌐 [Open data files](https://microsoft.github.io/finops-toolkit/data) were all updated.

<br>

Thanks to @helderpinto, @nteyan, @MSBrett, @ro100e, @aromano2, @Misba-Yousuf, @arthurclares, and @flanakin for their contributions!

Thanks to @didayal-msft, @sebassem, @SCuffy, @chris-bowman, @CaddyG, @jamelachahbar, @thecloudman, and @Springstone for their support!

---

**Comment by psilantropy** [comment]
Hi team.

Just curious if there was already a feature request to use an external variable file to handle the storage URL, scope, months, data source type etc across the multiple powerBI reports. Sounds like it's possible to use an excel workbook as a variable source.

It would make upgrades and configuring the multiple reports a lot quicker. Per user these variables shouldn't really change. I never keep those details in a quickly accessible place 😮‍💨

**Reply by flanakin** [reply]
Hmm, I haven't heard about that. Definitely open to trying that. Do you want to test it out and submit a GitHub issue to let us know how you'd expect to see it implemented?

Another approach I've heard about is to deploy a data-only report with all data sources and have other reports connect to it. This comes up in our conversations, but it hasn't bubbled up as a top priority yet. My only concern with this approach is that it'll likely require a separate set of reports, which would double our maintenance effort and make it harder for people to just test things out in Power BI Desktop. 

If there are any Power BI experts out there, I'd love to know how we could improve the deployment story. Power BI doesn't offer the same automation options that Azure does unfortunately.

**Reply by psilantropy** [reply]
To be honest it's not a big job to update the storage account link and agreement number. I think another report is probably a stretch.
There will always be something to update, and I can't really think of anything hugely easier than what has to be done already.

**Comment by saleemmy** [comment]
Hi Michael
What is the upgrade process from 0.4 to 0.5 or 0.6 without disrupting any changes in my modified powerbi reports? Thanks

**Reply by lundejd** [reply]
I am curious about this as well. I've gone through 4 different resource groups running versions of the FinOps toolkit because I don't trust simply re-running the deployment script. Has anyone done this? I have way too much to lose as I do the internal billing for 2.5M of spend a month and cannot have something break.

**Reply by SCuffy** [reply]
We're pretty good at notifying if there are any breaking changes, but then we don't know what you've customized. Paging @flanakin for comment.

**Reply by flanakin** [reply]
0.4 to 0.5 is an in-place upgrade that won't impact anything.
0.6 is introducing a change that will require the latest reports and older reports won't work against it. 
Let me add a page that documents the compatibility between hubs and Power BI reports.

**Reply by flanakin** [reply]
@psilantropy / @lundejd / @saleemmy, let me know if this helps: https://github.com/microsoft/finops-toolkit/pull/1028

Please leave comments and sign off to let me know if everything looks good.
